extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocation.java

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.api.SortColumn;
+import io.deephaven.base.verify.Require;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.BasicDataIndex;
+import io.deephaven.engine.table.impl.locations.*;
+import io.deephaven.engine.table.impl.locations.impl.AbstractTableLocation;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocation;
+import io.deephaven.parquet.table.location.ParquetTableLocationKey;
+import org.apache.iceberg.FileFormat;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class IcebergTableLocation implements TableLocation {
```

Comment:
I'm not clear that we need this class. Or if we do, I think it should extend `ParquetTableLocation` and implement `IcebergTableLocation`.